x-pack/legacy/plugins/lens/public/indexpattern_plugin/dimension_panel/field_select.tsx

```diff
@@ -5,37 +5,64 @@
  */
 
 import _ from 'lodash';
-import React, { useState } from 'react';
+import React, { useRef, useEffect } from 'react';
 import { i18n } from '@kbn/i18n';
-import { EuiComboBox, EuiButtonEmpty, EuiButtonIcon, EuiFlexItem } from '@elastic/eui';
-import { IndexPatternColumn, FieldBasedIndexPatternColumn } from '../indexpattern';
-import { IndexPatternDimensionPanelProps } from './dimension_panel';
-import { changeColumn, deleteColumn, hasField, sortByField } from '../state_helpers';
+import { EuiComboBox } from '@elastic/eui';
+import classNames from 'classnames';
+import {
+  IndexPatternColumn,
+  FieldBasedIndexPatternColumn,
+  OperationType,
+  BaseIndexPatternColumn,
+} from '../indexpattern';
+import { hasField, sortByField } from '../state_helpers';
 
-export interface FieldSelectProps extends IndexPatternDimensionPanelProps {
-  selectedColumn: IndexPatternColumn;
+export interface FieldSelectProps {
+  incompatibleSelectedOperationType: OperationType | null;
+  selectedColumn?: IndexPatternColumn;
   filteredColumns: IndexPatternColumn[];
+  onChangeColumn: (newColumn: IndexPatternColumn) => void;
+  onDeleteColumn: () => void;
 }
 
 export function FieldSelect({
+  incompatibleSelectedOperationType,
   selectedColumn,
   filteredColumns,
-  state,
-  columnId,
-  setState,
+  onChangeColumn,
+  onDeleteColumn,
 }: FieldSelectProps) {
-  const [isFieldSelectOpen, setFieldSelectOpen] = useState(false);
+  const inputRef = useRef<HTMLInputElement | null>(null);
   const fieldColumns = filteredColumns.filter(hasField) as FieldBasedIndexPatternColumn[];
 
   const uniqueColumnsByField = sortByField(
     _.uniq(
       fieldColumns
-        .filter(col => selectedColumn && col.operationType === selectedColumn.operationType)
+        .filter(col =>
+          incompatibleSelectedOperationType
+            ? col.operationType === incompatibleSelectedOperationType
+            : selectedColumn && col.operationType === selectedColumn.operationType
+        )
         .concat(fieldColumns),
       col => col.sourceField
     )
   );
 
+  useEffect(
+    () => {
+      if (!selectedColumn && incompatibleSelectedOperationType && inputRef.current) {
+        inputRef.current.focus();
+      }
+    },
+    [selectedColumn, incompatibleSelectedOperationType]
```

Comment:
I see what you're getting at with this behavior, but I actually found it really confusing as a user. I didn't like that the field selector covered the button I had just clicked because it prevented me from seeing the effect of my action.

Reply:
Hm, I see what you mean. I wanted to save an extra click here, but for new users it's indeed very confusing because you won't see the error message and everything. I'll remove it for now.